### PR TITLE
Add automatic module names for Spring Cloud Azure modules

### PR DESCRIPTION
**Pull request by hui1110**
As title.

Add the relevant configuration of `Automatic-Module-Name`to the pom file of each module of Spring Cloud Azure.

### CODE REVIEW

File: sdk/spring/spring-messaging-azure-eventhubs/pom.xml [hunk 1]
```diff
@@ -152,6 +152,21 @@
...
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jar-plugin</artifactId>
+        <version>3.1.2</version> <!-- {x-version-update;org.apache.maven.plugins:maven-jar-plugin;external_dependency} -->
+        <configuration>
+          <archive>
+            <manifestEntries>
+              <Automatic-Module-Name>com.azure.spring.messaging.eventhubs</Automatic-Module-Name>
+            </manifestEntries>
+            <manifest>
+              <addDefaultImplementationEntries>true</addDefaultImplementationEntries>
```
Comment (saragluna) [inline sdk/spring/spring-messaging-azure-eventhubs/pom.xml:165]: do we need this for a non-starter package?

File: sdk/spring/spring-cloud-azure-starter-integration-eventhubs/pom.xml [hunk 2]
```diff
@@ -101,4 +101,24 @@
...
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jar-plugin</artifactId>
+        <version>3.1.2</version> <!-- {x-version-update;org.apache.maven.plugins:maven-jar-plugin;external_dependency} -->
+        <configuration>
+          <archive>
+            <manifestEntries>
+              <Automatic-Module-Name>com.azure.spring.integration.eventhubs</Automatic-Module-Name>
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-starter-integration-eventhubs/pom.xml:113]: This is not correct, this module is a starter. should use this one `com.azure.spring.cloud.starter.integration.eventhubs`

File: sdk/spring/spring-cloud-azure-starter-integration-servicebus/pom.xml [hunk 3]
```diff
@@ -103,4 +103,24 @@
...
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jar-plugin</artifactId>
+        <version>3.1.2</version> <!-- {x-version-update;org.apache.maven.plugins:maven-jar-plugin;external_dependency} -->
+        <configuration>
+          <archive>
+            <manifestEntries>
+              <Automatic-Module-Name>com.azure.spring.integration.servicebus</Automatic-Module-Name>
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-starter-integration-servicebus/pom.xml:115]: Same here. `com.azure.spring.cloud.starter.integration.servicebus`

File: sdk/spring/spring-cloud-azure-starter-integration-storage-queue/pom.xml [hunk 4]
```diff
@@ -106,4 +106,24 @@
...
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jar-plugin</artifactId>
+        <version>3.1.2</version> <!-- {x-version-update;org.apache.maven.plugins:maven-jar-plugin;external_dependency} -->
+        <configuration>
+          <archive>
+            <manifestEntries>
+              <Automatic-Module-Name>com.azure.spring.integration.storage.queue</Automatic-Module-Name>
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-starter-integration-storage-queue/pom.xml:118]: same here, `com.azure.spring.cloud.starter.integration.storage.queue`

**Review by saragluna — CHANGES_REQUESTED** [review]

**Review by saragluna — APPROVED** [review]
LGTM.

File: sdk/spring/spring-cloud-azure-starter-active-directory-b2c/pom.xml [hunk 5]
```diff
@@ -141,4 +141,24 @@
...
   </dependencies>
 
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jar-plugin</artifactId>
+        <version>3.1.2</version> <!-- {x-version-update;org.apache.maven.plugins:maven-jar-plugin;external_dependency} -->
+        <configuration>
+          <archive>
```
Comment (moarychan) [inline sdk/spring/spring-cloud-azure-starter-active-directory-b2c/pom.xml:151]: Should we keep the same usage with the SDK parent configuration? SDK uses another attribute `<archive combine.children="append">`.

Comment (hui1110) [reply sdk/spring/spring-cloud-azure-starter-active-directory-b2c/pom.xml:151]: Perhaps we want some children to use a common resource directory and also define additional ones. For this, we can append the parent configuration by using `combine.children=append` on the resources element in the parent. You can control how child POMs inherit configuration from parent POMs by adding attributes to the children of the configuration element. But `Spring Cloud Azure Starters` will not be used as a parent POM and there is no parent POM now. This is [documented in the pom reference](https://maven.apache.org/pom.html#Plugins).

Comment (saragluna) [reply sdk/spring/spring-cloud-azure-starter-active-directory-b2c/pom.xml:151]: I don't think we need this, the sdk parent pom is a parent pom and expcts other pom to extend it. The same situation will never happens to our starter's poms.

**Pull request merged**